packages/wonder-blocks-link/components/__tests__/link.test.js

```diff
@@ -5,83 +5,310 @@ import {MemoryRouter, Route, Switch} from "react-router-dom";
 import {mount, unmountAll} from "../../../../utils/testing/mount.js";
 import Link from "../link.js";
 
+const wait = (delay: number = 0) =>
+    new Promise((resolve, reject) => {
+        // eslint-disable-next-line no-restricted-syntax
+        return setTimeout(resolve, delay);
+    });
+
 describe("Link", () => {
     beforeEach(() => {
         unmountAll();
     });
 
-    test("client-side navigation", () => {
-        // Arrange
-        const wrapper = mount(
-            <MemoryRouter>
-                <div>
-                    <Link testId="link" href="/foo">
-                        Click me!
-                    </Link>
-                    <Switch>
-                        <Route path="/foo">
-                            <div id="foo">Hello, world!</div>
-                        </Route>
-                    </Switch>
-                </div>
-            </MemoryRouter>,
-        );
-
-        // Act
-        const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
-        buttonWrapper.simulate("click", {button: 0});
-
-        // Assert
-        expect(wrapper.find("#foo").exists()).toBe(true);
-    });
+    describe("client-side navigation", () => {
+        test("works for known URLs", () => {
+            // Arrange
+            const wrapper = mount(
+                <MemoryRouter>
+                    <div>
+                        <Link testId="link" href="/foo">
+                            Click me!
+                        </Link>
+                        <Switch>
+                            <Route path="/foo">
+                                <div id="foo">Hello, world!</div>
+                            </Route>
+                        </Switch>
+                    </div>
+                </MemoryRouter>,
+            );
+
+            // Act
+            const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+            buttonWrapper.simulate("click", {button: 0});
+
+            // Assert
+            expect(wrapper.find("#foo").exists()).toBe(true);
+        });
+
+        test("navigation with unknown URL fails", () => {
```

Comment:
Nitpick: I find this test name a bit confusing, since it's really about route-based rendering and not navigation (which I associate more with the URL in the URL bar). Suggestion: "navigation to URL without route does not render".

Reply:
That's fair.  I'll change this to be more accurate.